### ISSUE TITLE
Rails Error: Unable to access log file. 

### ISSUE THREAD

**Issue by kswope** [phusion/passenger-docker#110]
I don't know why rails/passenger is creating a root owned log production.log file.  Using this container is the first time I've seen this.

Actually I wont be logging to production, I'm using the 12factor gem, but it looks like an unwritable production log screws up the process before anything can log to stdout and it all ends up in stdout.
Not using the 12factor gem gets me the same results so its not that. [EDIT] The 12factor gem is rightfully logging to stdout.  So really the only issue I have now is the error message about being unable to log to a file I wouldn't be logging to anyway, but its still annoying :)


```
App 81 stderr: Rails Error: Unable to access log file. Please ensure that /home/app/webapp/log/production.log exists and is writable (ie, make it writable for user and group: chmod 0664 /home/app/webapp/log/production.log). The log level has been raised to WARN and the output directed to STDERR until the problem is fixed.
```

```
root@711ad2fe8c1d:/home/app/webapp# ll -d log/
drwxrwxr-x 2 app app 4096 Oct 10 23:57 log//
root@711ad2fe8c1d:/home/app/webapp# ll log
total 8
drwxrwxr-x  2 app  app  4096 Oct 10 23:57 ./
drwxr-xr-x 25 app  app  4096 Oct 10 23:57 ../
-rw-rw-r--  1 app  app     0 Oct 10 23:54 .keep
-rw-r--r--  1 root root    0 Oct 10 23:57 production.log
```



```
   server {                                                                                                                                          
                                                                                                                                                     
       location ~ ^/assets/ {                                                                                                                        
         expires 1y;                                                                                                                                 
         add_header Cache-Control public;                                                                                                            
         add_header ETag "";                                                                                                                         
         break;                                                                                                                                      
       }                                                                                                                                             
                                                                                                                                                     
       listen 80;                                                                                                                                    
       server_name myapp;                                                                                                                   
       root /home/app/webapp/public;                                                                                                                 
                                                                                                                                                     
       passenger_enabled on;                                                                                                                         
       passenger_user app;                                                                                                                           
                                                                                                                                                     
       passenger_ruby /usr/bin/ruby;                                                                                                                 
                                                                                                                                                     
   }   





```

**Comment by FooBarWidget** [comment]
Do you happen to run any rails rake commands from the dockerfile? If so,  those commands are run as root and will create a root owned log file.

**Comment by kswope** [comment]
I've got

```
RUN chown -R app.app $rails_root
RUN RAILS_ENV=production rake assets:precompile
```

Looks like I just need to reverse the order of those RUNs and see what happens.  If this is true how isn't everybody getting bitten by this?

**Comment by mech** [comment]
I have this issues also. In my Dockerfile I don't have any rake task RUNing. However, I do have the `RUN whenever -w` does that count?

**Comment by mech** [comment]
Should I need to change to `USER` before running `whenever` like this:

```
USER app
RUN whenever -w
```

Or it should be like this

```
RUN sudo -u app whenever -w
```

**Comment by FooBarWidget** [comment]
passenger-docker (and baseimage-docker, on which it is based) provides the `setuser` tool for this which is better than both the `USER` stanza and `sudo`. Use it like this:

    RUN setuser app rake asset:precompile RAILS_ENV=production

See the baseimage-docker documentation for more information about `setuser`.

**Comment by kswope** [comment]
I'm starting to wonder if just skip all the USER and setuser and just put this at the end
```
   RUN chown -R app.app $rails_root                                                                                                                  
```

I don't think it creates another image layer and because COPY is so dumb (always copies as root!, just found that out) you are going to need a chown in there anyway.

**Comment by przbadu** [comment]
That one line `rake assets:precompile` screwed me up. Man I did used `chown`, even used `chmod -R 777 log/` and nothing worked. Previously it was fixed using sudo, but with latest release `0.9.19` I can't even use `sudo`.

And the fix was as @FooBarWidget  suggested, we can use `RUN setuser app` instead of `RUN sudo -u app`. Thanks a lot.

@FooBarWidget: rake task should be `assets:precompile`, i saw you are using `asset`, and rest are working perfectly fine.

`RUN setuser app rake assets:precompile RAILS_ENV=production`

Thanks
